File: implementations/java/cf_58a_chat_room.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class cf_58a_chat_room {
    public static void main(String[] args) throws Exception {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        String s = in.readLine();
        int lastIndice = 0;
        boolean hello = true;
        for(char letra : "hello".toCharArray()){
            lastIndice = s.indexOf(letra, lastIndice);
            if(lastIndice == -1){
                hello = false;
                break;
            }
            lastIndice++;
        }
        System.out.println(hello? "YES": "NO");
    }
}

File: implementations/cpp/cf_58a_chat_room.cpp
#include <bits/stdc++.h>

#define nl "\n"

using namespace std;

int main() {
    string s; cin>>s;
    size_t lastIndice = 0;
    bool helloFound = true;
    string hello = "hello";
    for(char letra : hello){
        lastIndice = s.find(letra,lastIndice);
        if(lastIndice == string::npos) {
            helloFound = false;
            break;
        }
        lastIndice++;
    }
    cout << (helloFound? "YES" : "NO") << nl;
}